huaweicloud/resource_huaweicloud_cbr_policy.go

```diff
@@ -0,0 +1,313 @@
+	)
+	if mErr.ErrorOrNil() != nil {
+		return mErr
+	}
+	if operationDefinition.MaxBackups != -1 {
+		d.Set("max_backups", operationDefinition.MaxBackups)
+	}
+	if operationDefinition.RetentionDurationDays != -1 {
+		d.Set("retention_duration_days", operationDefinition.RetentionDurationDays)
+	}
+
+	return nil
+}
+
+func resourceCBRPolicyV3Update(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	client, err := config.CbrV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating Huaweicloud CBR client: %s", err)
+	}
+
+	var updateOpts policies.UpdateOpts
+
+	if d.HasChange("name") {
+		newName := d.Get("name")
+		updateOpts.Name = newName.(string)
+	}
+
+	if d.HasChange("enabled") {
+		enabled := d.Get("enabled").(bool)
+		updateOpts.Enabled = &enabled
+	}
+
+	if d.HasChange("trigger_pattern") {
+		pattern := resourceCBRPolicyV3TriggerPattern(d)
+		updateOpts.Trigger = &policies.Trigger{
+			Properties: policies.TriggerProperties{
+				Pattern: pattern,
+			},
+		}
+	}
+
+	if d.HasChanges("max_backups", "retention_duration_days", "destination_project_name") {
+		opDefinition := resourceCBRPolicyV3OpDefinition(d)
+		updateOpts.OperationDefinition = opDefinition
+	}
+
+	_, err = policies.Update(client, d.Id(), updateOpts).Extract()
+	if err != nil {
+		return fmt.Errorf("error updating Huaweicloud CBR policy: %s", err)
+	}
+
+	return resourceCBRPolicyV3Read(d, meta)
+}
+
+func resourceCBRPolicyV3Delete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	client, err := config.CbrV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating Huaweicloud CBR client: %s", err)
+	}
+
+	log.Printf("[DEBUG] Deleting CBRv3 policy %s", d.Id())
+
+	if err = policies.Delete(client, d.Id()).ExtractErr(); err != nil {
+		return fmt.Errorf("eror deleting Huaweicloud CBR policy: %s", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func resourceCBRPolicyV3OpDefinition(d *schema.ResourceData) *policies.PolicyODCreate {
+	maxBackups, ok1 := d.GetOk("max_backups")
+	durationDays, ok2 := d.GetOk("retention_duration_days")
+	policyODCreate := policies.PolicyODCreate{}
+	if destinationProjectID, ok3 := d.GetOk("destination_project_id"); ok3 {
+		log.Printf("destination_project_id is %s", destinationProjectID.(string))
+		log.Printf("destination_project_name is %s", d.Get("destination_project_name").(string))
+
+		policyODCreate.DestinationProjectID = destinationProjectID.(string)
+		policyODCreate.DestinationRegion = d.Get("destination_project_name").(string)
+	}
+	if !ok1 && !ok2 {
+		policyODCreate.MaxBackups = -1
+		policyODCreate.RetentionDurationDays = -1
+		return &policyODCreate
+	}
+	policyODCreate.MaxBackups = maxBackups.(int)
+	policyODCreate.RetentionDurationDays = durationDays.(int)
+
+	return &policyODCreate
+}
+
+func makeSchedule(frequency, backupType, duration, time string) string {
+	timeSlice := strings.Split(time, ":")
```

Comment:
we should check the length of timeSlice

Reply:
it's added and will check at validate function.